arch/x86/64/source/ArchMemory.cpp

```diff
@@ -196,9 +195,9 @@ const ArchMemoryMapping ArchMemory::resolveMapping(uint64 pml4, uint64 vpage)
 
   assert(pml4 < PageManager::instance()->getTotalNumPages());
   m.pml4 = (PageMapLevel4Entry*) getIdentAddressOfPPN(pml4);
-  m.pdpt = 0;
-  m.pd = 0;
-  m.pt = 0;
+  m.pdpt = nullptr;
```

Comment:
also not sure this is a good change either. we're in an educational OS kernel --> the goal here is that students understand that this is nothing more than a number really

Reply:
I think there are plenty of other instances where that is demonstrated (e.g. syscalls). The explicit zero initialization of individual members here should just be replaced by explicit initialization of the struct variable.

`ArchMemoryMapping m;`  -> `ArchMemoryMapping m{};`